Modules/_cursesmodule.c

```diff
@@ -716,9 +754,14 @@ PyCursesWindow_New(WINDOW *win, const char *encoding)
 static void
 PyCursesWindow_Dealloc(PyCursesWindowObject *wo)
 {
-    if (wo->win != stdscr) delwin(wo->win);
-    if (wo->encoding != NULL)
+    if (wo->win != stdscr && wo->win != NULL) {
+        delwin(wo->win);
+        wo->win = NULL;
+    }
+    if (wo->encoding != NULL) {
```

Comment:
This check is not needed, PyMem_Free(NULL) is well defined: it does nothing.

Reply:
Ah, I see why I thought it wasn't. Actually, the online docs tell me that it's a no-op but the implementation does not tell me directly. I'll make the modifications tomorrow as well (actually I'll do everything tomorrow).